### ISSUE TITLE
Noob question on getting auth to work, invalid_grant

### ISSUE THREAD

**Issue by matthewdeanmartin** [halcy/Mastodon.py#286]
I have also posted this question to [stackoverflow](https://stackoverflow.com/questions/74631471/how-do-i-troubleshoot-a-invalid-grant-error-for-mastodon-mastodon-py). I'm 99% sure I am entering correct credentials and have tried this with multiple instances.

Thank you for writing this library, if I could only get past the front door, I see a lot of applications for mastodon.py.

(I also experimented with explicitly creating an app in the UI and attempting to use the credentials triplet from that, same invalid_grant on login.)

This code is taken almost verbatim from [mastodon.py's](https://mastodonpy.readthedocs.io/en/stable/) README.md

it always returns

> Traceback (most recent call last):   File
> "C:\Users\matth\GitHub\feed_thing\again.py", line 32, in <module>
>     mastodon.log_in(   File "C:\Users\matth\.virtualenvs\feed_thing-LXMa84iN\lib\site-packages\mastodon\Mastodon.py",
> line 568, in log_in
>     raise MastodonIllegalArgumentError('Invalid user name, password, or redirect_uris: %s' % e)
> mastodon.Mastodon.MastodonIllegalArgumentError: Invalid user name,
> password, or redirect_uris: ('Mastodon API returned error', 400, 'Bad
> Request', 'invalid_grant')

I have tried many variations, 3 different instances.

    import os
    from mastodon import Mastodon
    
    USERNAME_AS_EMAIL = os.environ["USERNAME_AS_EMAIL"]
    PASSWORD = os.environ["PASSWORD"]
    
    APP_NAME = "APP_NAME"
    Mastodon.create_app(
         APP_NAME,
         api_base_url = 'https://mastodon.social',
         to_file = 'pytooter_clientcred.secret'
    )
    mastodon = Mastodon(
        client_id = 'pytooter_clientcred.secret',
        api_base_url = 'https://mastodon.social'
    )
    mastodon.log_in(
        USERNAME_AS_EMAIL,
        PASSWORD,
        to_file = 'pytooter_usercred.secret'
    )
    mastodon = Mastodon(
        client_id = 'pytooter_clientcred.secret',
        access_token = 'pytooter_usercred.secret',
        api_base_url = 'https://mastodon.social'
    )

**Comment by halcy** [comment]
I can see nothing wrong with your code, and it in fact seems to work for me (using current state of master branch) verbatim:

http://aka-san.halcy.de/share/Untitled.ipynb_-_JupyterLab__Firefox_Developer_Ed_2022-11-30_19-11-04.png

What version are you using, and where did you install it from? I wonder if it could be some weird Windows issue (I develop on Windows, but inside WSL)

**Comment by matthewdeanmartin** [comment]
Okay I found the problem. The API code doesn't work with 2-factor auth. I have 2 factor auth enabled for my mastodon.social account, but not the others.

I turned of 2 factor auth )-; and the python code runs to completion.

--- troubleshooting info
I tried `mstdn.social` and `ohai.social` again and for those two, it works! Same code, I made sure to delete the *.secret files.

HOWEVER, when I switch back to mastadon.social - I'm back to `invalid_grant` -- mastodon social may have increased their security. 

(also, fwiw, I'm on Windows 10, python 3.9.7 ... I don't think this matters:)

```
$ pip freeze
blurhash==1.1.4
certifi==2022.9.24
charset-normalizer==2.1.1
decorator==5.1.1
idna==3.4
Mastodon.py==1.7.0
python-dateutil==2.8.2
python-magic==0.4.27
pytz==2022.6
requests==2.28.1
six==1.16.0
urllib3==1.26.13
```
